### PR TITLE
aws-sam-cli: 1.53.0 -> 1.78.0

### PR DESCRIPTION
**Pull request by junjihashimoto**
###### Description of changes

Resolve security issues and add new features that haven't been updated in a while.
https://github.com/aws/aws-sam-cli/releases

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/cfn-lint/default.nix [hunk 1]
```diff
@@ -19,13 +19,13 @@
...
   pname = "cfn-lint";
-  version = "0.73.2";
+  version = "0.76.1";
 
   src = fetchFromGitHub {
     owner = "aws-cloudformation";
     repo = "cfn-python-lint";
     rev = "refs/tags/v${version}";
-    hash = "sha256-CNB5LrXllGxy99NjCrbjkUXUpJ72U3pUnWqrqkOiCG8=";
+    hash = "sha256-CNB5LrXllGxy99NjCrbjkUXUpJ72U3pUnWqrqkOiCG9=";
```
Comment (waynevanson) [inline pkgs/development/python-modules/cfn-lint/default.nix:28]: The diff here is one character: `9` as second last character.

Will need to confirm this, as this unlikely to be the case.

File: pkgs/development/tools/aws-sam-cli/default.nix [hunk 2]
```diff
@@ -37,14 +40,6 @@ python3.pkgs.buildPythonApplication rec {
...
   patches = [
-    # Click 8.1 removed `get_terminal_size`, recommending
-    # `shutil.get_terminal_size` instead.
-    # (https://github.com/pallets/click/pull/2130)
-    ./support-click-8-1.patch
-    # Werkzeug >= 2.1.0 breaks the `sam local start-lambda` command because
-    # aws-sam-cli uses a "WERKZEUG_RUN_MAIN" hack to suppress flask output.
-    # (https://github.com/cs01/gdbgui/issues/425)
-    ./use_forward_compatible_log_silencing.patch
   ];
```
Comment (waynevanson) [inline pkgs/development/tools/aws-sam-cli/default.nix:43]: If these batches aren't used, can we remove the patch files?

File: pkgs/development/tools/aws-sam-cli/default.nix [hunk 3]
```diff
@@ -5,11 +5,11 @@
 
 python3.pkgs.buildPythonApplication rec {
   pname = "aws-sam-cli";
-  version = "1.53.0";
+  version = "1.78.0";
```
Comment (waynevanson) [inline pkgs/development/tools/aws-sam-cli/default.nix:8]: I've pulled in the 4 commits from this branch into a local branch and running `sam --version` outputted `SAM CLI, version 1.53.0`.

Still investigating to see if that's a fault of this PR or not.

**Review by waynevanson — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/aws-sam-cli/default.nix (hunk 3 above)
Comment (bluskript) [reply pkgs/development/tools/aws-sam-cli/default.nix:8]: I tested this PR locally, and correctly got version 1.78.0 as a version.
```
nix-shell -p '(import ./default.nix {}).aws-sam-cli'
❯ sam --version
SAM CLI, version 1.78.0
```

File: pkgs/development/tools/aws-sam-cli/default.nix (hunk 2 above)
Comment (junjihashimoto) [reply pkgs/development/tools/aws-sam-cli/default.nix:43]: This can not be patched, so I erased it, but I need to check it carefully.

**Comment by Smona** [comment]
Anything I can do to help move this forward? I view this as a pretty important PR, given this adds support for the node 18 runtime to `sam local`, and node 16 is marked unsafe in nixpkgs 23, which we're all supposed to be upgrading to right now.

**Comment by Artturin** [comment]
aws-sam-cli is at 1.90.0

**Comment by junjihashimoto** [comment]
@Artturin Thx!

**Pull request closed without merging**